pkg/internal/testhelpers/containers.go

```diff
@@ -143,12 +144,22 @@ func StartPGContainerWithImage(ctx context.Context, image string, testDataDir st
 		SkipReaper: false, /* switch to true not to kill docker container */
 	}
 
-	if testDataDir != "" {
-		req.BindMounts = map[string]string{
-			testDataDir: "/testdata",
+	if withExtension {
+		req.Cmd = []string{
+			"-c", "shared_preload_libraries=timescaledb",
+			"-c", "local_preload_libraries=pgextwlist",
+			"-c", "extwlist.extensions=timescale_prometheus_extra,timescaledb",
```

Comment:
This bugs me; there's a non-local dependency between the image being used and the `withExtension` flag where the flag needs to set iff the extension contains `pgextwlist` and `timescaledb`. If it wasn't for the fact that we need to use multiple versions for the upgrade tests, I'd be inclined to hardcode the images:tag and dispatch purely based on `withExtension`. Even in the current version, I think we should consider hardcoding the image we're looking at, dispatched by `withExtension`, and only take in the tag portion as an argument. (This would also make it easier to update the image we use, if we ever choose to do so. Right now it's written in a couple of different places)